### PR TITLE
Add partition locations if they are different from the table location

### PR DESCRIPTION
**Pull request by yuzhu**
This PR uses PathTranslator as a table-level cache for the path translation. So that we do not need to query the mount table repeatedly for path translations.

### CODE REVIEW

**Comment by yuzhu** [comment]
@gpang PTAL

File: core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java [hunk 1]
```diff
@@ -447,6 +447,16 @@ public void rename(AlluxioURI src, AlluxioURI dst, RenamePOptions options)
     });
   }
 
+  @Override
+  public AlluxioURI reverseResolve(AlluxioURI ufspath) throws IOException, AlluxioException {
+    checkUri(ufspath);
```
[the remark this answers is not in the document]
Comment (yuzhu) [reply core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java]: done

**Comment by yuzhu** [comment]
@gpang PTAL, thanks.

**Comment by yuzhu** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**